src/TestingModule.tsx

```diff
@@ -144,50 +187,86 @@ export const TestingModuleDescription = ({ api, running }: TestingModuleDescript
     [openVisualTestsPanel, warning]
   );
 
-  const emit = useChannel(
-    {
-      [TESTING_MODULE_RUN_ALL_REQUEST]: ({ providerId }) => {
-        if (providerId === TEST_PROVIDER_ID) startBuild();
-      },
-      [TESTING_MODULE_CANCEL_TEST_RUN_REQUEST]: ({ providerId }) => {
-        if (providerId === TEST_PROVIDER_ID) stopBuild();
-      },
-    },
-    [startBuild, stopBuild]
-  );
+  let description: string | React.ReactNode;
+  switch (true) {
+    case !!warning:
+      description = <Link onClick={clickWarning}>{warning}</Link>;
+      break;
+    case testProviderState === 'test-provider-state:running':
+      description = localBuildProgress
+        ? BUILD_STEP_CONFIG[localBuildProgress.currentStep].renderProgress(localBuildProgress)
+        : 'Starting...';
+      break;
+    case !!isOutdated:
+      description = 'Test results outdated';
+      break;
+    case localBuildProgress?.currentStep === 'aborted':
+      description = 'Aborted by user';
+      break;
+    case localBuildProgress?.currentStep === 'complete':
+      description = localBuildProgress.errorCount
+        ? `Encountered ${pluralize('component error', localBuildProgress.errorCount, true)}`
+        : warningStatusCount
+          ? `Found ${pluralize('story', warningStatusCount, true)} with ${pluralize('change', warningStatusCount)}`
+          : 'No visual changes detected';
+      break;
+    default:
+      description = 'Not run';
+  }
```

Comment:
this was essentially what the previous `TestingModuleDescription` component used to return. now we're setting it instead, and returning the JSX that `LegacyRender` would return - but simplified as it doesn't need to be generalized anymore.